library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -454,6 +509,14 @@ public long freeMemory(long windowId) throws IOException
         entryIter.remove();
 
         for (Map.Entry<Slice, BucketedValue> entry : windowData.entrySet()) {
+          /**
+           * The data still in memory and reachable before the memory released
+           * So put key into bloom filter here
+           */
+          if (bloomFilter != null) {
```

Comment:
I agree. Can we use `getFromReaders` in that case to insert into bloom filter? I don't think `freeMemory` is the right place to put values in the bloom filter.